packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts

```diff
@@ -1507,6 +1516,734 @@ describe('paybackloan for dusd using dfi', () => {
+    const dusdPaybackByTsla = new BigNumber(tslaPaybackAmount).multipliedBy(effectiveDusdPerTsla)
+
+    // Price of btc to dusd depends on the oracle, in this case 1 BTC = 1000 DUSD
+    const effectiveDusdPerBtc = new BigNumber(1000).multipliedBy(1 - btcPenaltyRate) // (DUSD per BTC * (1 - penalty rate))
+    const dusdPaybackByBtc = new BigNumber(btcPaybackAmount).multipliedBy(effectiveDusdPerBtc)
+
+    const dusdPayback = dusdPaybackByBtc.plus(dusdPaybackByTsla)
+
+    const dusdLoanRemainingAfter = new BigNumber(dusdLoanAmount).minus(dusdPayback).plus(dusdInterestAmountBefore)
+    const dusdInterestPerBlockAfter = dusdLoanRemainingAfter.multipliedBy(netInterest).dividedBy(365 * blocksPerDay)
+
+    // Let some time, generate blocks
+    await testing.generate(2)
+
+    const totalTslaPenalty = new BigNumber(tslaPaybackAmount).multipliedBy(tslaPenaltyRate)
+    const totalBtcPenalty = new BigNumber(btcPaybackAmount).multipliedBy(btcPenaltyRate)
+    const burnInfoAfter = await testing.rpc.account.getBurnInfo()
+    expect(burnInfoAfter.tokens).toStrictEqual([`${new BigNumber(tslaPaybackAmount).toFixed(8)}@TSLA`, `${new BigNumber(btcPaybackAmount).toFixed(8)}@BTC`])
+    expect(burnInfoAfter.dfipaybackfee).toStrictEqual(new BigNumber(0))
+    expect(burnInfoAfter.dfipaybacktokens).toStrictEqual([])
+    expect(burnInfoAfter.paybackfees).toStrictEqual([`${totalTslaPenalty.toFixed(8)}@TSLA`, `${totalBtcPenalty.toFixed(8)}@BTC`])
+    expect(burnInfoAfter.paybacktokens).toStrictEqual([`${dusdPayback.toFixed(8)}@DUSD`])
+
+    const blockHeightAfter = await testing.rpc.blockchain.getBlockCount()
+    const dusdInterestAmountAfter = dusdInterestPerBlockAfter.multipliedBy(blockHeightAfter - paybackLoanBlockHeight).decimalPlaces(8, BigNumber.ROUND_CEIL)
+    const dusdLoanAmountAfter = dusdLoanRemainingAfter.plus(dusdInterestAmountAfter)
+
+    const vaultAfter = await testing.rpc.loan.getVault(vaultId) as VaultActive
+    expect(vaultAfter.loanAmounts).toStrictEqual([`${dusdLoanAmountAfter.toFixed(8)}@DUSD`])
+    expect(vaultAfter.interestAmounts).toStrictEqual([`${dusdInterestAmountAfter.toFixed(8)}@DUSD`])
+  })
+
+  it('should be able to payback DUSD loan using DUSD - use PaybackLoanMetadataV2.loans', async () => {
+    // create DUSD-DFI
+    await testing.poolpair.create({
+      tokenA: 'DUSD',
+      tokenB: 'DFI'
+    })
+    await testing.generate(1)
+
+    // add DUSD-DFI
+    await testing.poolpair.add({
+      a: { symbol: 'DUSD', amount: 1000 },
+      b: { symbol: 'DFI', amount: 1000 }
+    })
+    await testing.generate(1)
+
+    const dusdInfo = await testing.rpc.token.getToken('DUSD')
+    const dusdId: string = Object.keys(dusdInfo)[0]
+    const paybackKey = `v0/token/${dusdId}/loan_payback/${dusdId}`
+    const penaltyRateKey = `v0/token/${dusdId}/loan_payback_fee_pct/${dusdId}`
+    const penaltyRate = 0 // Penaly rate will be always 0% as you are paying by dusd
```

Comment:
This actually not considered when DUSD loan paid by DUSD. It's  the same as TSLA paid by TSLA